src/components/Table/table.tsx

```diff
@@ -9,28 +10,61 @@ import styles from './styles.module.scss';
 
 interface Props {
   columns?: string[];
-  width?: number;
   children: string;
 }
 
-export const Table = ({ columns = [], children, width = 100 }: Props) => {
+export const Table = ({ columns = [], children }: Props) => {
   const [isExpand, setIsExpand] = useState(false);
-  const isExpandTable = width > 100;
+  const [isShowExpand, setIsShowExpand] = useState(false);
+  const [size, setSize] = useState(0);
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (ref && ref.current) {
+      const scrollHeight = ref?.current.offsetHeight - ref?.current.clientHeight;
+      setIsShowExpand(scrollHeight > 0);
+    }
+  }, [size]);
+
   useLockBodyScroll(isExpand);
-  const className = `table-${uuidv4(6)}`;
-  const columnsWidth = columns.map((column, index) => `
-    .${className} td:nth-child(${index + 1}), .${className} th:nth-child(${index + 1}) {
-      width: ${column};
+
+  useEffect(() => {
+    const element = ref && ref.current;
+    function handleResize() {
+      if (element) {
+        setSize(element.getBoundingClientRect().width);
```

Comment:
I suggest calling `setIsShowExpand(ref?.current.offsetHeight - ref?.current.clientHeight > 0)` right here, to avoid writing additional `useEffect()` (at lines 22-27) and state at line 19